zerver/tests/test_auth_backends.py

```diff
@@ -1100,6 +1138,26 @@ def stage_two_of_registration(self, result: HttpResponse, realm: Realm, subdomai
 
         self.assertFalse(user_profile.has_usable_password())
 
+    def verify_user_avatar(self, user: UserProfile, user_avatar_url: str='', use_social_avatar: bool=False,
+                           expected_social_avatar: BytesIO=None, response_social_avatar_hash: str='') -> None:
+        if user_avatar_url:
+            user_avatar_hash = user.avatar_hash
+            expected_social_avatar_hash = user_avatar_content_hash(expected_social_avatar.getvalue())
+            if use_social_avatar:
+                # Check if the uploaded content is correct.
+                self.assertEqual(response_social_avatar_hash, expected_social_avatar_hash)
+                # Check image from response is the same as the user avatar_hash.
+                self.assertEqual(response_social_avatar_hash, user_avatar_hash)
+                self.assertEqual(UserProfile.AVATAR_FROM_USER, user.avatar_source)
```

Comment:
I think the only checks that we should be making are `self.assertEqual(UserProfile.AVATAR_FROM_USER, user.avatar_source)` and `self.assertEqual(expected_social_avatar_hash, user_avatar_hash)`, I don't see why we want to check `response_social_avatar_hash`?  We can trust the `responses` library to return what we tell it. This will also allow to simplify this function a lot and get rid of a bunch of variables and arguments.